sdk/python/feast/inference.py

```diff
@@ -45,12 +48,72 @@ def infer_entity_value_type_from_feature_views(
                     entity.value_type != ValueType.UNKNOWN
                     and entity.value_type != inferred_value_type
                 ) or (len(extracted_entity_name_type_pairs) > 1):
-                    raise ValueError(
+                    raise RegistryInferenceFailure(
+                        "Entity",
                         f"""Entity value_type inference failed for {entity_name} entity.
-                        Multiple viable matches. Please explicitly specify the entity value_type
-                        for this entity."""
+                        Multiple viable matches.
+                        """,
                     )
 
                 entity.value_type = inferred_value_type
 
     return entities
+
+
+def infer_event_timestamp_column_for_data_sources(
+    data_sources: List[DataSource],
+) -> List[DataSource]:
+    ERROR_MSG_PREFIX = "Unable to infer DataSource event_timestamp_column"
+
+    for data_source in data_sources:
+        if (
+            data_source.event_timestamp_column is None
+            or data_source.event_timestamp_column == ""
+        ):
```

Comment:
Should we also verify that this timestamp column actually exists in the schema or not here?

Reply:
hmmm i think it should be done separately as a validate_data_source or something since there r other fields set in the datasource that may benefit from validation too. If it were the case that one passes a ts_column that doesn't exist and we'd want to infer it, then i think it would make more sense to have it here. But i think if it doesn't exist in the schema, we probably want to just throw exception rather than inferring it. It'd be nice if in the case that this method weren't called anymore, the only functionality lost is the inference, not any other needed checks.

Reply:
Fair, I think we can make a different PR for this validation.